### PR TITLE
libhttpseverywhere: Fix test for Linuxbrew

### PR DESCRIPTION
**Pull request by rwhogg**

### CODE REVIEW

**Comment by rwhogg** [comment]
Test failed (https://travis-ci.org/Linuxbrew/homebrew-core/builds/220420995#L7002)

```
/home/linuxbrew/.linuxbrew/Cellar/libhttpseverywhere/0.4.1/include/httpseverywhere-0.4/httpseverywhere.h:12:25: fatal error: libxml/tree.h: No such file or directory
```

The test apparently needs `libxml2`, but I'd have to look up the dependencies of all the formulae involved to see why it's not available at test time.

Also, I saw a reference to `-lintl` which is also wrong.

**Comment by rwhogg** [comment]
Now it passes on Circle but fails on Travis? https://travis-ci.org/Linuxbrew/homebrew-core/builds/220591053#L6999-L7018

```
/home/linuxbrew/.linuxbrew/bin/ld: cannot find -lhttpseverywhere-0.4

collect2: error: ld returned 1 exit status

Error: libhttpseverywhere: failed

Failed executing: /home/linuxbrew/.linuxbrew/bin/gcc-5 test.c -o test -Os -w -pipe -march=native -isystem/home/linuxbrew/.linuxbrew/include -L/home/linuxbrew/.linuxbrew/lib -Wl,--dynamic-linker=/home/linuxbrew/.linuxbrew/lib/ld.so -Wl,-rpath,/home/linuxbrew/.linuxbrew/lib -I/home/linuxbrew/.linuxbrew/opt/gettext/include -I/home/linuxbrew/.linuxbrew/opt/glib/include/glib-2.0 -I/home/linuxbrew/.linuxbrew/opt/glib/lib/glib-2.0/include -I/home/linuxbrew/.linuxbrew/Cellar/libhttpseverywhere/0.4.1/include/httpseverywhere-0.4 -I/home/linuxbrew/.linuxbrew/opt/json-glib/include/json-glib-1.0 -I/home/linuxbrew/.linuxbrew/opt/libarchive/include -I/home/linuxbrew/.linuxbrew/opt/libgee/include/gee-0.8 -I/home/linuxbrew/.linuxbrew/opt/libsoup/include/libsoup-2.4 -I/home/linuxbrew/.linuxbrew/opt/pcre/include -I/home/linuxbrew/.linuxbrew/opt/libxml2/include/libxml2 -D_REENTRANT -L/home/linuxbrew/.linuxbrew/opt/gettext/lib -L/home/linuxbrew/.linuxbrew/opt/glib/lib -L/home/linuxbrew/.linuxbrew/opt/json-glib/lib -L/home/linuxbrew/.linuxbrew/opt/libarchive/lib -L/home/linuxbrew/.linuxbrew/opt/libgee/lib -L/home/linuxbrew/.linuxbrew/opt/libsoup/lib -L/home/linuxbrew/.linuxbrew/Cellar/libhttpseverywhere/0.4.1/lib -larchive -lgee-0.8 -lgio-2.0 -lglib-2.0 -lgobject-2.0 -lhttpseverywhere-0.4 -ljson-glib-1.0 -lsoup-2.4 -lxml2

/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1836:in `block in system'

/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1774:in `open'

/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1774:in `system'

/home/linuxbrew/.linuxbrew/Library/Taps/homebrew/homebrew-core/Formula/libhttpseverywhere.rb:82:in `block in <class:Libhttpseverywhere>'

/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1673:in `block (2 levels) in run_test'

/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:884:in `with_logging'

/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1672:in `block in run_test'

/home/linuxbrew/.linuxbrew/Library/Homebrew/extend/fileutils.rb:14:in `block in mktemp'

/home/linuxbrew/.linuxbrew/Library/Homebrew/extend/fileutils.rb:74:in `block in run'

/home/linuxbrew/.linuxbrew/Library/Homebrew/extend/fileutils.rb:74:in `chdir'

/home/linuxbrew/.linuxbrew/Library/Homebrew/extend/fileutils.rb:74:in `run'

/home/linuxbrew/.linuxbrew/Library/Homebrew/extend/fileutils.rb:13:in `mktemp'

/home/linuxbrew/.linuxbrew/Library/Homebrew/formula.rb:1666:in `run_test'

/home/linuxbrew/.linuxbrew/Library/Homebrew/test.rb:33:in `block in <main>'

/home/travis/.rvm/rubies/ruby-2.0.0-p648-clang/lib/ruby/2.0.0/timeout.rb:66:in `timeout'

/home/linuxbrew/.linuxbrew/Library/Homebrew/test.rb:32:in `<main>'
```

File: Formula/libhttpseverywhere.rb [hunk 1]
```diff
@@ -58,6 +59,9 @@ def install
       -I#{libgee.opt_include}/gee-0.8
       -I#{libsoup.opt_include}/libsoup-2.4
       -I#{pcre.opt_include}
+    ]
+    flags << "-I#{libxml2.opt_include}/libxml2" unless OS.mac?
+    flags += %W[
```
Comment (sjackman) [inline Formula/libhttpseverywhere.rb]: Add
```ruby
flags << "-L#{Formula["libhttpseverywhere"].lib}"
```
?

**Comment by sjackman** [comment]
Thanks for fixing this test! It's been annoying me.

**Comment by maxim-belkin** [comment]
Travis:
```
/home/linuxbrew/.linuxbrew/bin/ld: cannot find -lhttpseverywhere-0.4
collect2: error: ld returned 1 exit status
Error: libhttpseverywhere: failed
```

**Comment by sjackman** [comment]
It looks as though on TravisCI it's not building the shared library `libhttpseverywhere-0.4.so`, though I can't say why.

**Comment by maxim-belkin** [comment]
Travis:
```
Possible undeclared dependencies:
  libxml2
```

**Comment by maxim-belkin** [comment]
Contents of `libhttpseverywhere` on Mac (note location of libraries):
```
$ brew list -v libhttpseverywhere
/usr/local/Cellar/libhttpseverywhere/0.4.2/.brew/libhttpseverywhere.rb
/usr/local/Cellar/libhttpseverywhere/0.4.2/include/httpseverywhere-0.4/httpseverywhere.h
/usr/local/Cellar/libhttpseverywhere/0.4.2/INSTALL_RECEIPT.json
/usr/local/Cellar/libhttpseverywhere/0.4.2/lib/girepository-1.0/HTTPSEverywhere-0.4.typelib
/usr/local/Cellar/libhttpseverywhere/0.4.2/lib/libhttpseverywhere-0.4.0.dylib
/usr/local/Cellar/libhttpseverywhere/0.4.2/lib/libhttpseverywhere-0.4.dylib
/usr/local/Cellar/libhttpseverywhere/0.4.2/lib/pkgconfig/httpseverywhere-0.4.pc
/usr/local/Cellar/libhttpseverywhere/0.4.2/LICENSE
/usr/local/Cellar/libhttpseverywhere/0.4.2/README.md
/usr/local/Cellar/libhttpseverywhere/0.4.2/share/libhttpseverywhere/rulesets.json
/usr/local/Cellar/libhttpseverywhere/0.4.2/share/vala/vapi/httpseverywhere-0.4.deps
/usr/local/Cellar/libhttpseverywhere/0.4.2/share/vala/vapi/httpseverywhere-0.4.vapi
```

**Comment by maxim-belkin** [comment]
Contents of Linux bottle:
```
$ tree
.
└── 0.4.2
    ├── include
    │   └── httpseverywhere-0.4
    │       └── httpseverywhere.h
    ├── INSTALL_RECEIPT.json
    ├── lib
    │   └── x86_64-linux-gnu
    │       ├── girepository-1.0
    │       │   └── HTTPSEverywhere-0.4.typelib
    │       ├── libhttpseverywhere-0.4.so -> libhttpseverywhere-0.4.so.0
    │       ├── libhttpseverywhere-0.4.so.0 -> libhttpseverywhere-0.4.so.0.4.2
    │       ├── libhttpseverywhere-0.4.so.0.4.2
    │       └── pkgconfig
    │           └── httpseverywhere-0.4.pc
    ├── LICENSE
    ├── README.md
    └── share
        ├── libhttpseverywhere
        │   └── rulesets.json
        └── vala
            └── vapi
                ├── httpseverywhere-0.4.deps
                └── httpseverywhere-0.4.vapi
```

**Comment by maxim-belkin** [comment]
Travis is not picking up the latest commits. Will re-submit this in a new PR.

**Comment by maxim-belkin** [comment]
see #2437

**Pull request closed without merging**